app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt

```diff
@@ -1,38 +1,56 @@
 package com.github.HumanLearning2021.HumanLearningApp.firestore
 
-import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import android.net.Uri
 import com.github.HumanLearning2021.HumanLearningApp.model.Category
 import com.github.HumanLearning2021.HumanLearningApp.model.DatasetInterface
 import com.google.firebase.FirebaseApp
 import com.google.firebase.firestore.DocumentId
 import com.google.firebase.firestore.DocumentReference
-import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.ktx.firestore
 import com.google.firebase.ktx.Firebase
 import com.google.firebase.ktx.app
+import com.google.firebase.storage.ktx.storage
 import kotlinx.coroutines.tasks.await
-import java.io.Serializable
+import java.util.*
 
 class FirestoreDatasetInterface(
+    /**
+     * name of a database within the Firebase App
+     */
```

Comment:
What do you mean by "within the Firebase App"?

Reply:
They are indeed different. My situation is that there can be separate databases in the same Firebase project, like test and production. (Currently we have 2 different test databases, see firebase-setup.md) The "DatasetInterface" needs to select one. The name of a dataset is associated with a dataset, which is an entity within the database. However, we currently operate under the assumption that there is only one dataset in a database. (see #50)

Reply:
Then I would suggest renaming the class to `FirestoreDatabaseInterface`